test/unit/model/authorization/TestTakerAuthorizationServiceTest.php

```diff
@@ -0,0 +1,300 @@
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ *
+ * Copyright (c) 2019 (original work) Open Assessment Technologies SA;
+ *
+ */
+
+namespace oat\taoProctoring\test\unit\model\authorization;
+
+use core_kernel_classes_Property;
+use core_kernel_classes_Resource;
+use Exception;
+use oat\generis\model\data\Ontology;
+use oat\generis\test\MockObject;
+use oat\generis\test\TestCase;
+use oat\oatbox\user\User;
+use oat\taoDelivery\model\authorization\UnAuthorizedException;
+use oat\taoProctoring\model\authorization\TestTakerAuthorizationService;
+use oat\taoProctoring\model\delivery\DeliverySyncService;
+use oat\taoDelivery\model\execution\DeliveryExecution;
+
+class TestTakerAuthorizationServiceTest extends TestCase
+{
+    /**
+     * @var Ontology|MockObject
+     */
+    private $ontologyMock;
+
+    /**
+     * @var TestTakerAuthorizationService
+     */
+    private $service;
+
+    protected function setUp()
+    {
+        parent::setUp();
+        $this->ontologyMock = $this->getMock(Ontology::class);
+        $this->service = new TestTakerAuthorizationService();
+    }
+
+    /**
+     * @dataProvider isActiveUnSecureDeliveryDataProvider
+     * @param string $propertyValue
+     * @param string $state
+     * @param bool $expected
+     * @throws Exception
+     */
+    public function testIsActiveUnSecureDelivery($propertyValue, $state, $expected)
+    {
+        $delivery = $this->getDeliveryMock();
+        $property = new core_kernel_classes_Property(
+            'http://www.tao.lu/Ontologies/TAODelivery.rdf#DeliveryTestRunnerFeatures'
+        );
+        $this->ontologyMock->expects($this->once())->method('getResource')->with('deliveryUri');
+
+        $this->ontologyMock->expects($this->once())
+            ->method('getProperty')
+            ->with('http://www.tao.lu/Ontologies/TAODelivery.rdf#DeliveryTestRunnerFeatures')
+            ->willReturn($property);
+
+        $delivery->expects($this->once())
+            ->method('getOnePropertyValue')
+            ->with($property)
+            ->willReturn($propertyValue);
+
+        $this->service->setServiceLocator($this->getServiceLocatorMock([Ontology::SERVICE_ID => $this->ontologyMock]));
+
+        if ($expected) {
+            $this->assertTrue($this->service->isActiveUnSecureDelivery('deliveryUri',$state));
+        } else {
+            $this->assertFalse($this->service->isActiveUnSecureDelivery('deliveryUri',$state));
+        }
+    }
+
+    /**
+     * @return array
+     */
+    public function isActiveUnSecureDeliveryDataProvider()
+    {
+        return [
+            'activeAndUnSecure' => [
+                null, 'http://www.tao.lu/Ontologies/TAODelivery.rdf#DeliveryExecutionStatusActive', true
```

Comment:
If you use names in dataProvider it's easier to understand your test:
```
'activeAndUnSecure' => [
    'propertyValue' => null,
    'executionState' => 'http://www.tao.lu/Ontologies/TAODelivery.rdf#DeliveryExecutionStatusActive',
    'expectedResult' => true
],
```

Reply:
fixed